footer.php

```diff
@@ -105,7 +105,7 @@ function draw_navbuttons() {
         if ($mod->sectionnum > 0 && $sectionnum != $mod->sectionnum) {
             $thissection = $sections[$mod->sectionnum];
 
-            if ($thissection->visible || !$COURSE->hiddensections ||
+            if ($thissection->visible || (property_exists($COURSE, "hiddensections") && !$COURSE->hiddensections) ||
```

Comment:
Hmm... I'll have to look at this in more detail, but I'd be wanting to use isset(), not property_exists (possibly also wanting to use empty(), but would have to figure out the logic of the line of code first anyway).